src/media-segment-request.js

```diff
@@ -235,59 +248,235 @@ const handleSegmentResponse = (segment, captionParser, finishProcessingFn) => (e
+    // really starts. To get around this, full segment appends should probe for the info
+    // needed.
+    const probeResult = probeTsSegment(bytes, segment.baseStartTime);
+
+    if (probeResult) {
+      trackInfoFn(segment, {
+        hasAudio: probeResult.hasAudio,
+        hasVideo: probeResult.hasVideo
+      });
+      trackInfoFn = null;
+
+      audioStartFn(probeResult.audioStart);
+      audioStartFn = null;
+      videoStartFn(probeResult.videoStart);
+      videoStartFn = null;
+    }
   }
 
-  // This is likely an FMP4 and has the init segment.
-  // Run through the CaptionParser in case there are captions.
-  if (segment.map && segment.map.bytes) {
+  transmux({
+    bytes,
+    transmuxer: segment.transmuxer,
+    audioAppendStart: segment.audioAppendStart,
+    gopsToAlignWith: segment.gopsToAlignWith,
+    isPartial,
+    onData: (result) => {
+      dataFn(segment, result);
+    },
+    onTrackInfo: (trackInfo) => {
+      if (trackInfoFn) {
+        trackInfoFn(segment, trackInfo);
+      }
+    },
+    onAudioTimingInfo: (audioTimingInfo) => {
+      // we only want the first start value we encounter
+      if (audioStartFn && typeof audioTimingInfo.start !== 'undefined') {
+        audioStartFn(audioTimingInfo.start);
+        audioStartFn = null;
+      }
+      // we want to continually update the end time
+      if (audioEndFn && typeof audioTimingInfo.end !== 'undefined') {
+        audioEndFn(audioTimingInfo.end);
+      }
+    },
+    onVideoTimingInfo: (videoTimingInfo) => {
+      // we only want the first start value we encounter
+      if (videoStartFn && typeof videoTimingInfo.start !== 'undefined') {
+        videoStartFn(videoTimingInfo.start);
+        videoStartFn = null;
+      }
+      // we want to continually update the end time
+      if (videoEndFn && typeof videoTimingInfo.end !== 'undefined') {
+        videoEndFn(videoTimingInfo.end);
+      }
+    },
+    onId3: (id3Frames, dispatchType) => {
+      id3Fn(segment, id3Frames, dispatchType);
+    },
+    onCaptions: (captions) => {
+      captionsFn(segment, [captions]);
+    },
+    onDone: (result) => {
+      // To handle partial appends, there won't be a done function passed in (since
+      // there's still, potentially, more segment to process), so there's nothing to do.
+      if (!doneFn || isPartial) {
+        return;
+      }
+      doneFn(null, segment, result);
+    }
+  });
+};
+
+const handleSegmentBytes = ({
+  segment,
+  bytes,
+  isPartial,
+  captionParser,
+  trackInfoFn,
+  timingInfoFn,
+  id3Fn,
+  captionsFn,
+  dataFn,
+  doneFn
+}) => {
+  const bytesAsUint8Array = new Uint8Array(bytes);
+
+  segment.isFmp4 =
+    // only set the property the first time we see some bytes so that partial appends
+    // don't try to check every section of bytes (since the check should only consider the
+    // first bytes in the segment)
+    typeof segment.isFmp4 === 'boolean' ? segment.isFmp4 :
+      isLikelyFmp4Data(bytesAsUint8Array);
```

Comment:
We should use `isLikelyFmp4Data` in all the places this is used

https://github.com/videojs/http-streaming/blob/de5aacb782f3f85754b77a7cd34c4e8a6e12ebcf/src/playlist.js#L478-L484